src/Illuminate/View/Concerns/AutowiresProperties.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Illuminate\View\Concerns;
+
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionProperty;
+use Illuminate\Support\Str;
+
+trait AutowiresProperties
+{
+    /**
+     * Automatically assign values to any properties on the class whose name
+     * exists in the supplied attributes, then call the mount() method.
+     *
+     * @return array
+     */
+    public function autowireProperties()
+    {
+        $class = new ReflectionClass($this);
+        $allAttributes = $this->attributes->getAttributes();
+
+        $properties = collect($class->getProperties(ReflectionMethod::IS_PUBLIC))
+            ->filter(function (ReflectionProperty $property) {
+                return !in_array($property->name, ['componentName', 'attributes']);
+            })
+            ->map(function (ReflectionProperty $property) {
+                return Str::camel($property->getName());
+            })
+            ->toArray();
+
+        foreach ($properties as $property) {
+            if (isset($this->attributes[$property])) {
+                $this->{$property} = $this->attributes[$property];
+                unset($this->attributes[$property]);
+            }
+        }
+
+        if (method_exists($this, 'mount')) {
+            app()->call([$this, 'mount'], $allAttributes);
```

Comment:
I'd recommend avoiding the `app()` helper here too - `Container::getInstance()` makes more sense.

Reply:
Both of those are kinda hacky, actually. Assumes there is only one container instance and it's shared with the entire app.

Reply:
While that sort of thing is fine in your own app code, and in the foundation namespace, it should be avoided in core components.

Reply:
Well, from looking through the code base, the `Container::getInstance()` approach is used in a lot of places.